src/fsverity/ioctl.rs

```diff
@@ -53,7 +62,11 @@ pub fn fs_ioc_enable_verity<F: AsFd, H: FsVerityHashValue>(fd: F) -> std::io::Re
                 sig_ptr: 0,
                 __reserved2: [0; 11],
             }),
-        )?;
+        ) {
+            Err(rustix::io::Errno::NOTTY) => return Err(EnableVerifyError::FilesystemNotSupported),
```

Comment:
Sorry for not mentioning that earlier, but there's also the case where the filesystem implementation supports the ioctl but the filesystem itself doesn't have it enabled.  This is the case for ext4, for example, where the `-O verity` flag was not specified to `mkfs.ext4` or `tune2fs`.  That's `EOPNOTSUPP`, I think:

https://www.kernel.org/doc/html/latest/filesystems/fsverity.html#fs-ioc-enable-verity

Reply:
Nice catch, thank you! I added this now and started writing a test but the test got a bit involved because it needs to create an ext4 image without verity, mount it, write a file on it and enable verity on it (which requires root to mount the ext4 image). So I did not incude it but I can clean it up and push that too if desired (it will probably need something like the mount.rs:TmpMount to cleanly umount on drop).